File: src/main/java/com/ByTech/ByTech/Cart/repositories/SaleRepository.java
package com.ByTech.ByTech.Cart.repositories;

import com.ByTech.ByTech.Cart.models.Sale;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import java.util.List;

public interface SaleRepository extends JpaRepository<Sale,Long> {
    List<Sale> findByClient_UserName(String userName);
    @Query("SELECT s FROM Sale s WHERE MONTH(s.date) = :month")
    List<Sale> findByMonth(@Param("month") int month);
}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/Sales/repositories/SaleRepository.kt
package com.example.ByTech_movil.Sales.repositories

import com.example.ByTech_movil.Sales.models.Sale
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface SaleRepository : JpaRepository<Sale, Long> {
    fun findByClient_UserName(userName: String): List<Sale>

    @Query("SELECT s FROM Sale s WHERE MONTH(s.date) = :month")
    fun findByMonth(@Param("month") month: Int): List<Sale>
}
